singularity-eos/closure/mixed_cell_models.hpp

```diff
@@ -1785,6 +1805,18 @@ PORTABLE_INLINE_FUNCTION SolverStatus PTESolver(System &s) {
     Real scale = s.ScaleDx();
     PORTABLE_REQUIRE(scale <= 1.0, "PTE Solver is attempting to increase the step size");
 
+    // If scale is very small, we may be iterating around the regime
+    // of validity and not making progress. Try to catch this, fail
+    // out early, and report this as a failure, so fallbacks can take
+    // over in a host code. We check that this has happened for at
+    // least two iterations, in case we can escape in one.
+    if ((std::abs(scale) < 1e-16) && (std::abs(scale_last) < 1e-16)) {
```

Comment:
Use a named constant, preferably something like `pte::ls::minimum_scale_magnitude`.

Reply:
Suggestion for this

```c++
ntry = 0;
/* ... */
if (std::abs (scale) < pte::min_scale ) ++ntry;
if (ntry > params.small_step_trys )
{ /* ...fails....*/ }
```

Reply:
Done.